src/source/Crypto/Tls_mbedtls.c

```diff
@@ -4,6 +4,35 @@
 #define LOG_CLASS "TLS_mbedtls"
 #include "../Include_i.h"
 
+// Read and parse CA certificate
+PRIVATE_API STATUS readAndParseCACertificate(PTlsSession pTlsSession)
+{
+    ENTERS();
+    STATUS retStatus = STATUS_SUCCESS;
+    UINT64 cert_len = 0;
+    PBYTE cert_buf = NULL;
+    CHAR errBuf[128];
+
+    CHK_STATUS(readFile(KVS_CA_CERT_PATH, FALSE, NULL, &cert_len));
+    CHK(cert_len > 0, STATUS_INVALID_CERT_PATH_LENGTH);
+    cert_buf = (PBYTE) MEMCALLOC(1, cert_len + 1);
+    CHK(cert_buf != NULL, STATUS_NOT_ENOUGH_MEMORY);
+    CHK_STATUS(readFile(KVS_CA_CERT_PATH, FALSE, cert_buf, &cert_len));
+    int ret = mbedtls_x509_crt_parse(&pTlsSession->cacert, cert_buf, (SIZE_T) (cert_len + 1));
```

Comment:
Suggest adding null check before dereferencing `pTlsSession`.

Reply:
Added the check:
`CHK(pTlsSession != NULL && pTlsSession->cacert != NULL, STATUS_NULL_ARG);`

Reply:
Thanks for updating.

The CI is failing with the new changes: https://github.com/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/actions/runs/15013768862/job/42187165938?pr=2121#step:6:5520

`cacert` isn't a pointer, so the null check is causing a compile error (not needed): `pTlsSession->cacert != NULL`. We can keep the `TlsSession != NULL` check though.

Reply:
Oh! Missed that. Updated. The `mbedtls_x509_crt_parse` API anyway checks for the pointer internally, so could have worked even if it was pointer.